bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java

```diff
@@ -0,0 +1,414 @@
+      }
+
+      if (inputFilePath != null) {
+        Preconditions.checkArgument(
+            zookeeperPort == null && zookeeperQuorum == null,
+            ZOOKEEPER_PORT_KEY
+                + "/"
+                + ZOOKEEPER_QUORUM_KEY
+                + " can not be set when input file is set.");
+        Preconditions.checkArgument(
+            tableNameFilter == null,
+            TABLE_NAME_FILTER_KEY
+                + " is not supported for reading the schema from a table. "
+                + "TableFilter should be used when writing the schema to the file.");
+      } else {
+        Preconditions.checkArgument(
+            zookeeperQuorum != null && zookeeperPort != null, "Schema source not specified. ");
+      }
+    }
+
+    public static SchemaTranslationOptions loadOptionsFromSystemProperties() {
+      SchemaTranslationOptions options = new SchemaTranslationOptions();
+      options.projectId = System.getProperty(PROJECT_ID_KEY);
+      options.instanceId = System.getProperty(INSTANCE_ID_KEY);
+      options.outputFilePath = System.getProperty(OUTPUT_FILE_KEY);
+      options.inputFilePath = System.getProperty(INPUT_FILE_KEY);
+      options.zookeeperQuorum = System.getProperty(ZOOKEEPER_QUORUM_KEY);
+      if (System.getProperty(ZOOKEEPER_PORT_KEY) != null) {
+        options.zookeeperPort = Integer.parseInt(System.getProperty(ZOOKEEPER_PORT_KEY));
+      }
+
+      options.tableNameFilter = System.getProperty(TABLE_NAME_FILTER_KEY);
+
+      // Ensure that the options are set properly
+      // TODO It is possible to validate the options without creating the object, but its less
+      // readable. See if we can make it readable and validate before calling the constructor.
+      try {
+        options.validateOptions();
+      } catch (Exception e) {
+        usage(e.getMessage());
+        throw e;
+      }
+
+      return options;
+    }
+  }
+
+  interface SchemaReader {
+
+    ClusterSchemaDefinition readSchema() throws IOException;
+  }
+
+  static class FileBasedSchemaReader implements SchemaReader {
+
+    String schemaFilePath;
+
+    public FileBasedSchemaReader(String schemaFilePath) {
+      this.schemaFilePath = schemaFilePath;
+    }
+
+    @Override
+    public ClusterSchemaDefinition readSchema() throws IOException {
+      return new ObjectMapper().readValue(new File(schemaFilePath), ClusterSchemaDefinition.class);
+    }
+  }
+
+  static class HBaseSchemaReader implements SchemaReader {
+    private final String tableFilterPattern;
+    private final Admin hbaseAdmin;
+
+    public HBaseSchemaReader(String zookeeperQuorum, int zookeeperPort, String tableFilterPattern)
+        throws IOException {
+
+      // If no filter is provided, use `.*` to match all the tables.
+      this.tableFilterPattern = tableFilterPattern == null ? ".*" : tableFilterPattern;
+
+      // Create the HBase admin client.
+      Configuration conf = HBaseConfiguration.create();
```

Comment:
Do you expect to inherit any values from hbase-site.xml? if not, pass false to create

Reply:
I set the bare minimum properties required to connect, but if there are additional properties in Hbase-site.xml, then we should load them. I would assume things like auth configuration to connect to the cluster etc.